### PR TITLE
Update MLI docstrings part 1

### PR DESCRIPTION
**Pull request by AlyssaCote**
I added and updated the MLI docstrings so that everything renders correctly and they are more consistent. This means I needed to make some decisions on what "consistent" means, because there were many different combinations going on. Feel free to push back on these! I just went with what looked like the best/what we were doing the most.

- The docstring description uses punctuation to end the description. 
- The `:param:`, `:returns:`, etc do not use punctuation to end their description line UNLESS the line needs multiple sentences, in which case, use punctuation. (I think this happens once or twice).
- All description lines are capitalized. 

example:
```py
def serialize_response(response: response_capnp.ResponseBuilder) -> bytes:
        """
        Serializes a built response message.

        :param response: Response to be serialized
        :return: Serialized response bytes
        """
        return response.to_bytes()
```

### CODE REVIEW

**Review by mellis13 — APPROVED** [review]
Only a couple of very small comments.  Thanks for fixing this up!

File: smartsim/_core/mli/infrastructure/control/device_manager.py [hunk 1]
```diff
@@ -124,10 +129,10 @@ def get_device(
         batch: RequestBatch,
         feature_stores: dict[str, FeatureStore],
     ) -> _GeneratorContextManager[WorkerDevice]:
-        """Get the device managed by this object
+        """Get the device managed by this object.
 
-        the model needed to run the batch of requests is
-        guaranteed to be available on the model
+        The model needed to run the batch of requests is
```
[the remark this answers is not in the document]
Comment (AlyssaCote) [reply smartsim/_core/mli/infrastructure/control/device_manager.py:134]: I think so! Fixed.

File: smartsim/_core/mli/infrastructure/control/device_manager.py [hunk 2]
```diff
@@ -109,9 +114,9 @@ def _load_model_on_device(
```
[the remark this answers is not in the document]
Comment (AlyssaCote) [reply smartsim/_core/mli/infrastructure/control/device_manager.py:114]: Fixed!

File: smartsim/_core/mli/infrastructure/control/device_manager.py [hunk 3]
```diff
@@ -46,37 +47,40 @@ def __init__(self, name: str) -> None:
...
     def __contains__(self, key: str) -> bool:
-        """Check if model with a given key is available on the device
+        """Check if model with a given key is available on the device.
 
-        :param key: the key of the model to check for existence
-        :returns: whether the model is available on the device
+        :param key: The key of the model to check for existence
+        :returns: Whether the model is available on the device
         """
         return key in self._models
```
[the remark this answers is not in the document]
Comment (AlyssaCote) [reply smartsim/_core/mli/infrastructure/control/device_manager.py:86]: Oops! Fixed. (Definitely check this one.)

**Pull request merged**